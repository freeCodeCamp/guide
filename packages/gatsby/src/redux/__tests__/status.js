const { actions, boundActions } = require(`../actions`)
const statusReducer = require(`../reducers/status`)

Date.now = jest.fn(() => 1482363367071)

describe(`Status actions/reducer`, () => {
  it(`allows setting plugin status`, () => {
    expect(
      actions.setPluginStatus({ something: `test status` })
    ).toMatchSnapshot()
  })

  it(`allows updating status`, () => {
    let state = statusReducer(
      undefined,
      actions.setPluginStatus({ test: `test status` }, { name: `test-plugin` })
    )
    state = statusReducer(
      state,
      actions.setPluginStatus({ test: `test status2` }, { name: `test-plugin` })
    )
    expect(state.plugins[`test-plugin`].test).toEqual(`test status2`)
  })

  it(`throws an error if status isn't an object`, done => {
    function runReducer() {
      return statusReducer(
        undefined,
        actions.setPluginStatus(`test job`, { name: `test-plugin` })
      )
    }

    expect(runReducer).toThrowErrorMatchingSnapshot()
    done()
  })

  it(`throws an error if the plugin name isn't set`, done => {
    function runReducer() {
      return statusReducer(
        undefined,
        actions.setPluginStatus({ blah: `test job` })
      )
    }

    expect(runReducer).toThrowErrorMatchingSnapshot()
    done()
  })
})
