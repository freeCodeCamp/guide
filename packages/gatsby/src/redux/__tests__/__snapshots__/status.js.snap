// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status actions/reducer allows setting plugin status 1`] = `
Object {
  "payload": Object {
    "something": "test status",
  },
  "plugin": undefined,
  "type": "SET_PLUGIN_STATUS",
}
`;

exports[`Status actions/reducer throws an error if status isn't an object 1`] = `"You must pass an object into setPluginStatus. What was passed in was \\"test job\\""`;

exports[`Status actions/reducer throws an error if the plugin name isn't set 1`] = `"Cannot read property 'name' of undefined"`;
